hw/ip/spi_host/doc/_index.md

```diff
@@ -0,0 +1,633 @@
+- Finally within a given byte, more significant bits are transmitted and received first.
+For Dual and Quad transactions the *most* significant bit in any instantaneous pair or nibble is transmitted or received on `sd[0]`, and the remaining `sd` bits (1 though 3) are populated in order of *decreasing* significance.
+
+The following figure shows how data appears on the serial data bus when it is written to TXDATA, or read from RXDATA.
+Within a particular byte, the most significant bit is always transmitted or received first.
+Meanwhile, the individual bytes are meant to be transmitted as they are laid out in processor memory.
+The order of the individual bytes varies with the `ByteOrder` parameter, so that for Big-Endian (or Little-Endian) systems, the most (or least) significant bytes are transmitted and/or received first.
+
+{{< wavejson >}}
+{signal: [
+  ["ByteOrder=0",
+  {name: "sd[0] (host output)", wave: "x22222222222|2222|222|22x", data: ["t[31]", "t[30]", "t[29]", "t[28]", "t[27]", "t[26]", "t[25]", "t[24]", "t[23]","t[22]",
+                                                                          "t[21]","t[17]","t[16]","t[15]","t[14]","t[8]", "t[7]", "t[6]", "t[1]", "t[0]"]},
+  {name: "sd[1] (host input)", wave: "x22222222222|2222|222|22x", data: ["r[31]", "r[30]", "r[29]", "r[28]", "r[27]", "r[26]", "r[25]", "r[24]", "r[23]","r[22]",
+                                                                         "r[21]","r[17]","r[16]","r[15]","r[14]","r[8]", "r[7]", "r[6]", "r[1]", "r[0]"]},
+  {name: "Which Byte?", wave: "x4.......4..|..4.|.4.|..x", data: ["MSB from TXDATA/RXDATA", "","", "          LSB"]}
+],
+  ["ByteOrder=1",
+  {name: "sd[0] (host output)", wave: "x22222222222|2222|222|22x", data: ["t[7]", "t[6]", "t[5]", "t[4]", "t[3]", "t[2]", "t[1]", "t[0]", "t[15]","t[14]",
+                                                                          "t[13]","t[9]","t[8]","t[23]","t[22]","t[16]", "t[31]", "t[30]", "t[25]", "t[24]"]},
+  {name: "sd[1] (host input)", wave: "x22222222222|2222|222|22x", data: ["r[7]", "r[6]", "r[5]", "r[4]", "r[3]", "r[2]", "r[1]", "r[0]", "r[15]","r[14]",
+                                                                         "r[13]","r[9]","r[8]","r[23]","r[22]","r[16]", "r[31]", "r[30]", "r[25]", "r[24]"]},
+  {name: "Which Byte?", wave: "x5.......5..|..5.|.5.|..x", data: ["LSB from TXDATA/RXDATA", "","", "          MSB"]}
+],
+  ],
+  head: {
+    text: "Serial bit ordering for 32-bit TXDATA & RXDATA data words (t[31:0] and r[31:0]), Standard SPI as a Function of the Parameter 'ByteOrder'",
+  },
+  foot: {
+  text: "(Bits are numbered as they appear when loaded into TXDATA or RXDATA registers.)"
+  }
+}
+{{< /wavejson >}}
+
+
+For Dual and Quad mode transfers similar time ordering applies, as shown in the following figure, though many bits of similar significance are packed into multiple parallel `sd` data lines, with the most significant going to `sd[0]` (*not* sd[3]).
```

Comment:
would you mind making this particular clear in the feature list or something? 
I can definitely see a system integrator accidentally hooking this up wrong.

Reply:
Well, this doesn't say how to hook the wires up.  SD[0] should go to sd{0] or IO[0] on the device.   This isn't about how to  hook up the device, it just says that the data ordering convention matches our target device.

I can add a sentence that says this though...
```
- Data ordering conventions match those commonly used by NOR flash devices such as XYZ and ABC
```
Is that what you hand in mind?

Reply:
i just meant making it clear very early on which bit is most significant. 
Because for example on the chip level they might be looking at MOSI / MISO lines, and it's possible for there to be mistakes.

Reply:
How about this:  I added this to the feature list:
- Support for Standard SPI, Dual SPI or Quad SPI commands
  - Pins `sd[0]` through `sd[3]` intended to respectively map to lines IO<sub>0</sub> through IO<sub>3</sub>.
  - Pin `sd[0]` may also be identified as "MOSI" by other SPI Hosts, while `sd[1]` is commonly referred to as "MISO"

Reply:
Wording changed subtly in text, please take a look.

Reply:
thanks!